### ISSUE TITLE
Add caching parameter to BatchDataLoader constructor

### ISSUE THREAD

**Issue by SebastianStehle** [graphql-dotnet/graphql-dotnet#3674]
The DataLoaderBase has a flag to indicate whether caching is allowed. This flag is not available in the BatchDataLoader class and in the extension methods.

Would be great if this flag would be allowed. I have a data loader scenario where I also define the fields I want to retrieve as part of the IDs. Then I merge all fields and make a single request. But the results are not cacheable because they might not have the necessary fields  for other queries.

**Comment by Shane32** [comment]
Great idea!  I'll look into it.

On a side note, I solve for this scenario in my own code in the following manner: also include the `GraphQLField` (`resolveFieldContext.FieldAst`) also as part of the id passed to the data loader (so the key is a ValueTuple of the AST and id).  Then within the data loader it performs a groupby on the AST node, in order to determine which ids need different sets of fields returned, and it uses the AST node for each group to determine which fields need to be returned for each set, and then runs different SQL queries/etc for each group based on the fieldset required.  Since the key contains the id and the AST node, results are cached appropriately (although the benefit of such a cache is doubtful).

**Comment by SebastianStehle** [comment]
Yes, I do basically the same with the AST. But I can query results across multiple data sets with a single query, therefore I use the "union" approach.
